app/server/controller/src/main/java/io/syndesis/server/controller/integration/online/PublishHandler.java

```diff
@@ -149,6 +149,7 @@ private DeploymentData createDeploymentData(Integration integration, Integration
             .addSecretEntry("application.properties", propsToString(applicationProperties));
 
         integration.getConfiguredProperties().forEach(deploymentDataBuilder::addProperty);
+        integration.getLabels().forEach((k, v) -> deploymentDataBuilder.addLabel(Labels.validateKey(k), Labels.validate(v)));
```

Comment:
Here we should expect labels to be valid, I don't think there is a need to re-validate here.